modules/sharethroughBidAdapter.js

```diff
@@ -1,123 +1,178 @@
-
-      const nonHttp = sharethroughInternal.getProtocol().indexOf('http') < 0;
-      query.secure = nonHttp || (sharethroughInternal.getProtocol().indexOf('https') > -1);
-
-      if (bidderRequest && bidderRequest.gdprConsent && bidderRequest.gdprConsent.consentString) {
-        query.consent_string = bidderRequest.gdprConsent.consentString;
-      }
-
-      if (bidderRequest && bidderRequest.gdprConsent) {
-        query.consent_required = !!bidderRequest.gdprConsent.gdprApplies;
-      }
-
-      if (bidderRequest && bidderRequest.uspConsent) {
-        query.us_privacy = bidderRequest.uspConsent
-      }
-
-      if (config.getConfig('coppa') === true) {
-        query.coppa = true
+    const timeout = config.getConfig('bidderTimeout');
+
+    const nonHttp = sharethroughInternal.getProtocol().indexOf('http') < 0;
+    const secure = nonHttp || (sharethroughInternal.getProtocol().indexOf('https') > -1);
+
+    const req = {
+      id: utils.generateUUID(),
+      at: 1,
+      cur: ['USD'],
+      tmax: timeout,
+      site: {
+        domain: window.location.hostname,
+        page: window.location.href,
+        ref: bidderRequest.refererInfo ? bidderRequest.refererInfo.referer || null : null,
+      },
+      user: {
+        ext: {
+          eids: userIdAsEids(bidRequests[0]),
+        },
+      },
+      device: {
+        ua: navigator.userAgent,
+        language: navigator.language,
+        js: 1,
+        dnt: navigator.doNotTrack === '1' ? 1 : 0,
+        h: window.screen.height,
+        w: window.screen.width,
+      },
+      regs: {
+        coppa: config.getConfig('coppa') === true ? 1 : 0,
+        ext: {},
+      },
+      source: {
+        ext: {
+          version: '$prebid.version$',
+          str: VERSION,
+          schain: bidRequests[0].schain,
+        },
+      },
+      bcat: bidRequests[0].params.bcat || [],
+      badv: bidRequests[0].params.badv || [],
+      test: 0,
+    };
+
+    if (bidderRequest.gdprConsent) {
+      const gdprApplies = bidderRequest.gdprConsent.gdprApplies === true;
+      req.regs.ext.gdpr = gdprApplies ? 1 : 0;
+      if (gdprApplies) {
+        req.user.ext.consent = bidderRequest.gdprConsent.consentString;
       }
+    }
 
-      if (bidRequest.schain) {
-        query.schain = JSON.stringify(bidRequest.schain);
-      }
+    if (bidderRequest.uspConsent) {
+      req.regs.ext.us_privacy = bidderRequest.uspConsent;
+    }
 
-      const floor = getFloor(bidRequest);
-      if (floor) {
-        query.bidfloor = floor;
-      }
+    const imps = bidRequests.map(bidReq => {
+      const impression = {};
 
-      if (bidRequest.params.badv) {
-        query.badv = bidRequest.params.badv;
+      const gpid = utils.deepAccess(bidReq, 'ortb2Imp.ext.data.pbadslot');
+      if (gpid) {
+        impression.ext = { gpid: gpid };
       }
 
-      if (bidRequest.params.bcat) {
-        query.bcat = bidRequest.params.bcat;
+      if (bidReq.mediaTypes && bidReq.mediaTypes.video) {
```

Comment:
You should check that the context for the video request is not outstream

Reply:
@ncolletti good idea, how would I go about doing this?

Reply:
@ncolletti added some logic to the `isBidRequestValid` method, to filter out video requests which are not outstream

Reply:
@epechuzal This may be problematic to include in `isBidRequestValid` for publishers who use multiple mediatypes in their request to your adapter. For example, if you were to include 1 banner and 1 video outstream for the same adunit, that request would fail and the imp object would not build for the banner request. If you move the outstream context check to the line above, 90, that will not occur.

Reply:
@ncolletti ok so I moved it to line 90 of `buildRequests`. If the context is outstream, then we just return `null`

Later will filter out any falsy `imps` so that we don't end up sending `null` to the adserver. 

Would this work?

Reply:
Yep this is good, thank you